### ISSUE TITLE
Lint/OrderedMagicComments reports at line is not a magic comment

### ISSUE THREAD

**Issue by ken1flan** [rubocop-hq/rubocop#6115]
## Expected behavior
Given the following code:

```
# frozen_string_literal: true

x = { encoding: Encoding::SJIS }
puts x
```

I expect RuboCop to throw no error.
Because line 3 is not a magic comment.

## Actual behavior

RuboCop throws an error:

```
test.rb:3:1: W: Lint/OrderedMagicComments: The encoding magic comment should precede all other magic comments.
```

## Steps to reproduce the problem

Create a file with the following contents:

```
# frozen_string_literal: true

x = { encoding: Encoding::SJIS }
puts x
```

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version

```
$ rubocop -V
0.58.1 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-darwin17)
```

**Comment by koic** [comment]
Thanks for the interesting feedback. There was a false positive to misunderstanding the Hash notation as an encoding magic comment.
I confirmed that and I opened a PR #6117. Thanks!
